$NetBSD: patch-configure,v 1.1 2025/01/31 12:22:44 micha Exp $

Add libtool support.
Upstream will not merge this (libtool is listed as not acceptable).
IRIX: plain manual pages, not pack(1)ed formatted ones.

--- configure.orig
+++ configure
@@ -1489,6 +1489,7 @@
   --enable-use_std        allow FLTK to use std::string etc.
   --disable-print         turn off print support (X11)
   --enable-shared         turn on shared libraries
+  --enable-libtool        pkgsrc: Use libtool
   --disable-svg           disable SVG support
   --disable-threads       turn off multi-threading support
   --enable-x11            use X11 with Cygwin or macOS (default=no)
@@ -5174,6 +5175,18 @@
 
 
 
+# pkgsrc: Check whether --enable-libtool or --disable-libtool was given.
+if test "${enable_libtool+set}" = set; then
+  enableval="$enable_libtool"
+
+if test x$enable_libtool != xno; then
+    LIBTOOL="$enable_libtool"
+    enable_shared=no
+else
+    LIBTOOL=""
+fi
+fi
+
 if test "x$enable_shared" = xyes
 then :
 
@@ -5484,7 +5497,25 @@
 
 
 
-
+# pkgsrc: libtool support
+if test "x$LIBTOOL" != "x"; then
+    LINKFLTK="../src/libfltk.la"
+    if test "x$LINKFLTKGL" != "x"; then
+        LINKFLTKGL="../src/libfltk_gl.la"
+    fi
+    if test "x$LINKFLTKIMG" != "x"; then
+        LINKFLTKIMG="../src/libfltk_images.la"
+    fi
+    if test "x$LINKFLTKCAIRO" != "x"; then
+        LINKFLTKCAIRO="../src/libfltk_cairo.la"
+    fi
+    LINKSHARED="-lfltk_images -lfltk_forms -lfltk"
+    DSONAME="libfltk.la"
+    FLDSONAME="libfltk_forms.la"
+    GLDSONAME="libfltk_gl.la"
+    IMGDSONAME="libfltk_images.la"
+    CAIRODSONAME="libfltk_cairo.la"
+fi
 
 
 
@@ -10409,10 +10440,11 @@
  ;; #(
   irix*) :
 
-    # SGI IRIX
-    CAT1EXT=z
-    CAT3EXT=z
-    CAT6EXT=z
+    # SGI IRIX: plain manual pages, as pkgsrc installs them (packed
+    # formatted pages need pack(1), which a cross build does not have)
+    CAT1EXT=1
+    CAT3EXT=3
+    CAT6EXT=6
  ;; #(
   *) :
 
